### ISSUE TITLE
InfoWindow causes jump panning when clicking markers

### ISSUE THREAD

**Issue by ndimatteo** [fullstackreact/google-maps-react#444]
If you pan the map and then click any marker (even ones contained in the current zoom view), the map will jump the center back to the default and then animate a pan to the active InfoWindow.

While this ensures the open InfoWindow is always visible, it disrupts the users desired panning and creates a terrible user experience with the map jumping all over the place.

See below for visual context:

![map-jump](https://user-images.githubusercontent.com/737188/82938401-becc7000-9f5f-11ea-925f-2905950982fd.gif)

**Comment by EricMontague** [comment]
I was having the same issue and figured out that this is caused by the improper comparisons between the center and currentLocation objects in the componentDidUpdate method in the source code of the Map component. The statements are currently written as:

`
    if (this.props.center !== prevProps.center) {
      this.setState({
        currentLocation: this.props.center
      });
    }
    if (prevState.currentLocation !== this.state.currentLocation) {
      this.recenterMap();
    }
  }`

and should instead be


`if (
      this.props.center.lat !== prevProps.center.lat ||
      this.props.center.lng !== prevProps.center.lng
    ) {
      
      this.setState({
        currentLocation: this.props.center
      });
    }
    if (
      prevState.currentLocation.lat !== this.state.currentLocation.lat ||
      prevState.currentLocation.lng !== this.state.currentLocation.lng
    ) {

      this.recenterMap();
    }`


This is because in Javascript, even if two objects have the same properties values, they will be treated as two different objects. Comparing the latitude and longitude properties directly should correct this issue. I hope this helps someone else who comes across this problem in the future.

**Comment by ndimatteo** [comment]
@EricMontague this is an awesome find!

Since I'm using this through NPM for a react-powered site (with builds happening through Netlify), it sounds like someone would need to update the core package in order for one to take advantage of this fix.

Not sure how to address this in the interim.

**Comment by EricMontague** [comment]
@ndimatteo You’re right! There’s also another bug caused by the same issue in the componentDidUpdate of the Marker component that causes markers to flicker when hovering over them.

I’m not sure this project is being maintained anymore, so even if we suggested these changes, they likely wouldn’t get added to this library.

I’m planning on forking this repo and making the changes I need to the source code in that forked version. Then I’m going to use npm to add my forked repo to my dependencies instead of this library. I think that’s probably our best bet.

**Comment by chacepro** [comment]
Any chance this was fixed?
